ForumProject/communications/consumers.py

```diff
@@ -0,0 +1,135 @@
+import json
+import logging
+from channels.generic.websocket import AsyncWebsocketConsumer
+
+from .utils import get_room, get_messages, get_online_users, get_user_first_name, \
+    add_user_to_online, remove_user_from_online, create_message
+
+
+class ChatConsumer(AsyncWebsocketConsumer):
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(args, kwargs)
+        self.room_name = None
+        self.room_group_name = None
+        self.room = None
+        self.messages = None
+        self.user = None
+        self.user_inbox = None
+
+    async def connect(self):
+        if not self.scope['user'].is_authenticated:
+            await self.close()
+            return
+
+        self.room_name = self.scope['url_route']['kwargs']['room_name']
+        self.room_group_name = f'chat_{self.room_name}'
+        self.room = await get_room(self.room_name)
+        self.messages = await get_messages(self.room)
+        self.user = self.scope['user']
+        self.user_inbox = f'inbox_{self.user.first_name}'
+
+        await self.accept()
```

Comment:
You're accepting the connection after performing some operations (e.g., getting the room). It’s usually better to accept the connection as soon as you determine the user is valid to ensure a responsive user experience.